src/components/Upload.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import Dropzone from 'react-dropzone';
+import { Field } from 'redux-form';
+import PropTypes from 'prop-types';
+import styles from './Upload.scss';
+
+const renderDropzone = field => (
+  <div>
+    <Dropzone
+      accept="image/*"
+      maxSize="3000"
+      className={styles.overide}
+      name={field.name}
+      onDrop={fileToUpload => field.input.onChange(fileToUpload)}
+    >
+      <div>
+        <p className={styles.header} >
+          <i className={`material-icons ${styles.logo}`}>publish</i> &nbsp; &nbsp; Click here to upload a file </p>
+        <p className={styles.body} >
+         Your file should not be bigger than 3 MB it can be .png .jpg .pdf
+        </p>
+      </div>
+    </Dropzone>
+    {field.input.value && (
+    <div>
+      { <img
+        style={styles.preview}
+        alt={field.input.name}
+        src={field.input.value[0].preview}
+      />} </div>)}
+  </div>);
+
+const Upload = ({ name, ...props }) => (
+  <Field
+    {...props}
+    name={name}
+    component={renderDropzone}
+  />
+    );
+Upload.propTypes = {
+  name: PropTypes.string.isRequired,
+};
+export default Upload;
+
+// TODO: add translation
+// TODO: add PDF support with special icon
```

Comment:
Ah, ok. The problem with PDFs is that they can not be used in an `<img src=` tag. So this is a situation where technical requirements are going to hurt the user experience, and we will have to make compromises.

@alxska We can either 1) reject pdf. 2) accept pdf but don't show preview or 3) spend two weeks building a pdf-rendering service that will show a preview after a few seconds delay.
Preference?